### PR TITLE
Add size settings to the `Product Image` block

### LINKED ISSUE #10024: Add size settings to the `Product Image` block
Currently, the `Product Image` block does not allow to change the size of the image, we need that to achieve different designs for new patterns.

The size setting should allow changes in width & height, similar to the ones in the `Post Featured Image` block.
Screenshot 2023-06-28 at 14 53 09

This issue also includes replacing the `Post Featured Image` block by the `Product Image` in the `Product hero` pattern after this setting is added.

### PR DESCRIPTION
**Pull request by albarin**
This PR adds size settings: width, height, and scale to the `Product Image` block.
This work is part of the block improvements needed in some blocks to make them more flexible to achieve the new pattern designs.

Also, in the `Product Hero` pattern, the `Product Featured Image` is replaced by the `Product Image` with the height setting set to 300px.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/10024

### Screenshots
Screenshot 2023-06-29 at 15 47 41

#### User-Facing Testing

**Product Image settings**
1. Create a new page or post.
2. Insert the `Single Product` block.
3. Click on the `Product Image` block and open the settings sidebar.
4. Check the `Image size` settings are shown in the sidebar.
5. Change the settings and check the changes are properly reflected on the editor and in the front end, after saving.

**Product Hero pattern**
1. Insert the `Product Hero` pattern.
2. Click the image and check the block used is the `Product Image`.
3. Save and check in the front end that the patterns renders fine.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Product Image block: add size settings to change the width, height, and scale.

### CODE REVIEW

File: assets/js/atomic/blocks/product-elements/image/image-size-settings.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+				} }
+				value={ width }
+				units={ [
+					{
+						value: 'px',
+						label: 'px',
+					},
+				] }
+			/>
+			{ height && (
```
Comment (kmanijak) [inline assets/js/atomic/blocks/product-elements/image/image-size-settings.tsx:83]: I think the Scale option could be available if either `height` or `width` is provided. Is there a reason to use only `height` as a condition?

**Review by kmanijak — COMMENTED** [review]
Great work!

I found one difference between editor and frontend when image is set to wider than available space. In Editor image shows in full scale, while in frontend it's cropped and horizontally centered. That may be discouraging to use by merchants if they see the content overlapping in editor. But I'm actually not sure which version should be expected one 🤔 

Editor | Frontend
-- | --
image | image

File: assets/js/atomic/blocks/product-elements/image/image-size-settings.tsx (hunk 1 above)
Comment (albarin) [reply assets/js/atomic/blocks/product-elements/image/image-size-settings.tsx:83]: I basically followed the same logic as in the `Post Featured Image` block in Gutenberg.

**Comment by albarin** [comment]
> Great work!
> 
> I found one difference between editor and frontend when image is set to wider than available space. In Editor image shows in full scale, while in frontend it's cropped and horizontally centered. That may be discouraging to use by merchants if they see the content overlapping in editor. But I'm actually not sure which version should be expected one 🤔

@kmanijak Thanks for the review, good catch! Not sure either which should be the expected one. I pushed a fix following again the `Post Feature Image` which allows the image to be wider than the available space. WDYT?

**Review by kmanijak — APPROVED** [review]
> I pushed a fix following again the Post Feature Image which allows the image to be wider than the available space. WDYT?

I found one thing a bit disturbing: the image is at the top of the content on the frontend, while it's below in the editor:

Editor | Frontend
-- | --
image | image

But I checked and Post Featured Image behaves exactly the same, so I think that deserves an upstream fix. I'm approving then, but what do you think about creating an issue in Gutenberg?

**Comment by albarin** [comment]
> But I checked and Post Featured Image behaves exactly the same, so I think that deserves an upstream fix. I'm approving then, but what do you think about creating an issue in Gutenberg?

Thanks @kmanijak! I fixed the issue for the `Product Image` in this PR and I'll open a new issue in GB 👌

**Pull request merged**